### ISSUE TITLE
DAMS statistics reporting is questionable and needs investigating

### ISSUE THREAD

**Issue by gamontoya** [ucsdlib/damsmanager#166]
## Special Collections and Archives:
- [Dr. Seuss Political Cartoons](https://library.ucsd.edu/dc/collection/bb65202085) had a HUGE spike from January to April: 109k object views on the exhibit pages in Feb alone, the DAMS is only reporting ~18k.
- [San Diego History collection](https://library.ucsd.edu/dc/collection/bb33815809) is still not showing up in the [DAMS Object Hits by Month](https://library.ucsd.edu/damsmanager/statsCollectionObjects.do)

## RDCP:
- Ho Jung reports that she is showing 0 hits (or downloads) for some objects.
    - Need example(s)

**Comment by gamontoya** [comment]
@hjsyoo Can you please provide me with an example or two?

**Comment by hjsyoo** [comment]
@gamontoya I haven't had time to do anything but trivial testing at this point, but I keep a brief record of the stats manager issues here: https://lisn.ucsd.edu/display/datacuration/DAMS+Statistics+development+timeline. See the latest entry for examples.

**Comment by lsitu** [comment]
@rstanonik: Is there any changes to the weblog /pub/data1/weblog/httpd located at lib-hydratail-staging? 
I tried to access some objects from dams4 on library but I don't see the URL is logged at /pub/data1/weblog/httpd. For example:
https://library.ucsd.edu/dc/object/bb6793765b.

**Comment by rstanonik** [comment]
It appears that https://library isn't logged to the central log server, only http://library.  I'll talk with John about it.

**Comment by lsitu** [comment]
Thanks @rstanonik .

**Comment by rstanonik** [comment]
John fixed logging so that https://library now logs to the central log server.  I tested by browsing https://library.ucsd.edu/dc/object/bb6793765b and saw my request logged in /pub/data1/weblog/httpd on lib-hydratail-staging.

**Comment by lsitu** [comment]
@jhriv  @rstanonik: Thank you very much.

**Comment by lsitu** [comment]
Hi Gabriela,
I think the RDCP issue from Ho Jung and the DAMS Object Hits by Month issue in "San Diego History collection" should be fixed with the weblog changes. Could we test it on prod to see whether there are other issues related?
For the 109k HUGE spike object views in "Dr. Seuss Political Cartoons had a HUGE spike from January to April on the exhibit pages, I think this should be the bot/spider counts. We filter out some of the bot/spider accesses at this time so I think ~18k from DAMS sounds like reasonable. Would you like to change the rules to count all views/access from the exhibit pages to make it consistent with the counts on the exhibit pages instead?

**Comment by gamontoya** [comment]
@lsitu Yes, I suspected the exhibits pages were reporting bot hits. As you suggest, please change the rules so they match in both places.

@hjsyoo Can you start visiting some of your collections/objects that were getting zero hits and then check again to see if the hits are being picked up now (on prod)?

**Comment by lsitu** [comment]
@gamontoya I've updated DAMS Statistics to include all counts from the exhibits pages, See PR https://github.com/ucsdlib/damsmanager/pull/168 .
@mcritchlow The PR is ready fro review now. Thanks.

**Comment by hjsyoo** [comment]
@gamontoya I'm running a small test today and will report tomorrow on the results. Do we know when  https://library no longer logged to the central log server? Did this only become an issue when the library site migration happened in June? I'm just trying to figure out when the stats began omitting actual hits.

**Comment by lsitu** [comment]
@jhriv @rstanonik Do you have any ideas about @hjsyoo's question regarding when the weblog issue happened?

**Comment by jhriv** [comment]
Part of the migration was a more aggressive use of https:// over http:// 

I don't have any records going back of https://library logging centrally.

**Comment by gamontoya** [comment]
@rstanonik @jhriv If http://library is a redirect (to https, or is it?) shouldn't it captured by the central log server?

Shouldn't **http:**//library.ucsd.edu/dc/object/bb6793765b = **https:**//library.ucsd.edu/dc/object/bb6793765b

When was the actual switch to https://

**Comment by gamontoya** [comment]
@mdpeters @hjsyoo Can you both report back on what you're seeing now for your stats?

**Comment by hjsyoo** [comment]
@gamontoya My small, controlled test at the end of August indicated that downloads are being logged correctly, but views are not being logged at all. I haven't had time to do a proper test of all user behavior on the various file types we have, but given our discussion of the D5 migration, think it's best to put this on hold until further discussion.

**Comment by lsitu** [comment]
@hjsyoo I see some stats on both the curator view and non-curactor views for August and September: https://library.ucsd.edu/damsmanager/statsRdcpUsage.do.

Could you give us more information regarding what objects were tested for the view and when the tests were performed so that we can look into it? Thanks.

**Comment by hjsyoo** [comment]
@gamontoya I can't explain why some numbers are >0 and others won't increment. I've put my notes into a google doc: https://docs.google.com/spreadsheets/d/1JWic6IF1e3yswqqNmOxxFy35lIwuhu7_Kbuvz62ynbs/edit#gid=0. The development timeline notes are here: https://lisn.ucsd.edu/display/datacuration/DAMS+Statistics+development+timeline. Again, I think this requires some more rigorous testing than I can provide at the moment. I'm hiring a student soon, whom I'm hoping to train to do some testing.

**Comment by lsitu** [comment]
@hjsyoo I guess there could be some cache issue with your browser in your tests. Could you open a new private browser /Incognito window each time when you start a test? Thanks.

**Comment by hjsyoo** [comment]
@lsitu How frequently should I open a new window? This isn't something I was aware would be an issue. If I want to test 3 objects, for example, and do 2 downloads of each, then I have to close the window/tab between each download, meaning I open a new incognito window 6x for this one test?

**Comment by lsitu** [comment]
@hjsyoo I think opening a new incognito window for each round of test for those 3 objects in your case will be fine. But if you want another round of test with the same objects to accumulate the counts of views, then we may need to open a new incognito window again. We just want to make sure that no caching will affect our tests.

**Comment by hjsyoo** [comment]
@lsitu I'm in the process of hiring a student, whom I'll train to do some more rigorous testing. It may take a few weeks before we can report anything. I'll plan to add caching as a factor in our tests.
Thanks.
